firmware/quadruna/VC/src/io/io_imu.c

```diff
@@ -0,0 +1,73 @@
+#include "io_imu.h"
+#include "hw_i2c.h"
+#include "app_canTx.h"
+#include <math.h>
+#include <stdio.h>
+
+extern I2C_HandleTypeDef hi2c2;
+static I2cInterface     *imu;
+
+// Default sensitivity for LSM6DSM is 0.061 mg/digit
+const float SENSITIVITY = 0.061f;
+
+bool io_imu_init()
+{
+    imu->i2c_handle     = &hi2c2;
+    imu->target_address = 0x6B;
+    imu->timeout_ms     = 100;
+
+    return hw_i2c_is_target_ready(imu);
+}
+
+bool io_imu_init_accelerometer()
+{
+    uint8_t buffer[1];
+    buffer[0] = 0x40; // turn on accelerometer to normal mode
+
+    return hw_i2c_mem_write(imu, 0x10, 1, buffer, sizeof(buffer));
+}
+
+void read_register(uint8_t *rx_buffer, uint16_t mem_addr)
+{
+    hw_i2c_mem_read(imu, mem_addr, 1, &rx_buffer[0], 1);         // LSB
+    hw_i2c_mem_read(imu, (mem_addr + 0x1), 1, &rx_buffer[1], 1); // MSB
+}
+
+double io_imu_acceleration_x()
+{
+    uint8_t x_data[2];
+    read_register(x_data, 0x28);
+    int16_t x_raw = (int16_t)((int16_t)x_data[1] << 8 | x_data[0]);
+
+    // Convert raw value to acceleration in m/s^2
+    float x_acceleration = x_raw * SENSITIVITY * 9.81f / 1000.0f;
+
+    app_canTx_VC_X_set(x_acceleration);
+    return x_acceleration;
+}
+
+double io_imu_acceleration_y()
+{
+    uint8_t y_data[2];
+    read_register(y_data, 0x2A);
+    int16_t y_raw = (int16_t)((int16_t)y_data[1] << 8 | y_data[0]);
```

Comment:
Instead of declaring a byte array, you can pass in a `uint16_t`. ARM microcontrollers (the ones we use!) are little endian, which means that least significant bytes have lower addresses, and more significant bytes have higher memory addresses. 

So, for example, if a `uint16_t` is stored at address 0x2000, then the LSB lives at 0x2000 and the MSB lives at 0x2001. Similarly, a `uint8_t` array of length 2 stored at 0x2000 will have element 0 at 0x2000 and element 1 at 0x2001. 

Notice that a regular `uint16_t`, and a `uint8_t[2]` with its LSB/MSB representing a 16-bit int, will have equivalent representations in memory (LSB at 0x2000, MSB at 0x2000 + 1).

Reply:
I removed the `read_register` function since it became redundant and I can just call `hw_i2c_mem_read` directly, but `hw_i2c_mem_read` requires a `uint8_t* rx_buffer` since it is easier to work with byte size data and I think most registers in `HAL_I2C` have one byte registers. So wouldn't it just be best to keep the `uint8_t y_data[2]` array instead?